client/src/main/java/org/asynchttpclient/request/body/multipart/InputStreamPart.java

```diff
@@ -0,0 +1,49 @@
+package org.asynchttpclient.request.body.multipart;
+
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
+import static org.asynchttpclient.util.Assertions.assertNotNull;
+
+public class InputStreamPart extends FileLikePart {
+
+  private final InputStream inputStream;
+  private final long contentLength;
+
+  public InputStreamPart(String name, InputStream inputStream, long contentLength, String fileName) {
```

Comment:
small nit: could you please add a constructir without contenLength that sets the value as `-1`? There's a chance most users won't know the stream length beforehand.